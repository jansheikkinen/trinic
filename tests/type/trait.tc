// Trait

function main(void) -> void do
  let a: trait Add = undefined;
end
