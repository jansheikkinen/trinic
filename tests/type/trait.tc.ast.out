## PARSER ##
Parsing tests/type/trait.tc...

### LEXER ###
Generating tokens from source...
Generated 17 tokens
000:     FUNCTION,     (null), (003, 008)
001:   IDENTIFIER,       main, (003, 013)
002:            (,     (null), (003, 014)
003:         VOID,     (null), (003, 018)
004:            ),     (null), (003, 019)
005:           ->,     (null), (003, 021)
006:         VOID,     (null), (003, 026)
007:           DO,     (null), (003, 029)
008:          LET,     (null), (004, 005)
009:   IDENTIFIER,          a, (004, 007)
010:            :,     (null), (004, 008)
011:        TRAIT,     (null), (004, 014)
012:   IDENTIFIER,        Add, (004, 018)
013:            =,     (null), (004, 020)
014:    UNDEFINED,     (null), (004, 030)
015:            ;,     (null), (004, 031)
016:          END,     (null), (005, 003)

### AST GENERATION ###
function main(VOID) -> VOID do
  let a: trait Add,    = UNDEFINED,   
  end

